src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php

```diff
@@ -226,7 +226,9 @@ public function canBeEnabled()
             ->beforeNormalization()
                 ->ifArray()
                 ->then(function ($v) {
-                    $v['enabled'] = isset($v['enabled']) ? $v['enabled'] : true;
+                    if (!isset($v['enabled'])) {
+                        $v['enabled'] = !empty($v);
```

Comment:
Should probably be `is_array($v) && !empty($v)` to cover the `foo: ~` case, which should work as enable.

Reply:
What tests should we add to make sure it won't happen in the future? Component tests cover `null` case already. `~` is just a `null`, right?

Reply:
forget my comment. This callback is only triggered for arrays ;)

Reply:
Looks like `~` in yaml is represented by empty array, simple reproducer is available [here](https://github.com/symfony/symfony/issues/25862)